h2o-core/src/main/java/hex/grid/filter/PermutationFilterFunction.java

```diff
@@ -0,0 +1,9 @@
+package hex.grid.filter;
+
+import hex.Model;
+
+import java.util.function.Predicate;
+
+public interface PermutationFilterFunction<MP extends Model.Parameters>
```

Comment:
Looks like you could get rid of this one: it has actually same contract as `FilterFunction`

Reply:
@seb-h2o I can see what you are trying to achieve with your last comments. It feels like "the end justifies the means" :) You want me to get rid of Activatable (include it into FilterFunction) and then reuse `FIlterFunction` for what now is `PermutationFilter`. 
I don't think I we should do this for couple of reasons:
   1) `PermutationFilter` and `FilterFunction` are different things. Former is working with functions, latter is working with permutations. Even though they all aimed at filtering the levels of abstractions are different.
  2) If we do what you suggested then we will extend Activatable for `PermutationFilter` which is another proof that these things are different. We don't have concept of activating at this level (set of functions). We can pretend that we can do activation but there is no real use case now. This is same as activating stateless function - we can define that activation just don't do anything for stateless but this is actually violation of OOP principles. Things should not care about things they are not applicable to.  It is like generalising over Cat and Ball - Ball.meow() can just return silence. 
3) And the most important point I think is that if we use `FilterFunction` interface for everything, we will be able to pass `KeepOnlyFirstMatchFilterFunction` to the constructor of the `RandomDiscreteValueWalker` directly and test would fail - as the logic for activation is outside of this type of filters. 
I can commit changes with the test if you want.

Reply:
I didn't suggest `PermutationFilter` == `FilterFunction`, I can accept the difference here,
but `PermutationFilterFunction` == `FilterFunction`

Reply:
@seb-h2o sorry! that was actually partially done. I have all interfaces already moved to `FilterFunction` specifically for that purpose - but forgot to remove/replace.